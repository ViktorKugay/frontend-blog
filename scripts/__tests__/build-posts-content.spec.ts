import {buildPostsContent} from '../build-posts-content';
import {resolve} from 'path';

const INPUT_POSTS_MOCK_PATH_VALID = resolve('scripts', '__tests__', 'mock-valid');
const INPUT_POSTS_MOCK_PATH_INVALID = resolve('scripts', '__tests__', 'mock-invalid');

describe('BuildContent', () => {
  describe('when posts has valid content', () => {
    it('should build valid json', () => {
      const buildedArticle = buildPostsContent(INPUT_POSTS_MOCK_PATH_VALID);
      expect(buildedArticle).toMatchSnapshot();
    });

    it('should has valid attributes', () => {
      const [buildedArticle] = buildPostsContent(INPUT_POSTS_MOCK_PATH_VALID);
      expect(buildedArticle.attributes.id).toBeDefined();
      expect(buildedArticle.attributes.title).toBeDefined();
      expect(buildedArticle.attributes.description).toBeDefined();
      expect(buildedArticle.attributes.image).toBeDefined();
      expect(buildedArticle.attributes.date).toBeDefined();
    });
  });

  describe('when post has no required attributes', () => {
    it('should throw assertion error', () => {
      try {
        buildPostsContent(INPUT_POSTS_MOCK_PATH_INVALID);
        expect('this code').toBe('never executed');
      } catch (e) {
        expect(e.message).toBe('ID');
      }
    });
  });
});
