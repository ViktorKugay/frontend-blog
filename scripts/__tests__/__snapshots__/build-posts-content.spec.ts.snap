// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildContent when posts has valid content should build valid json 1`] = `
Array [
  Object {
    "attributes": Object {
      "date": "2020-04-05T00:00:00.000Z",
      "description": "test_description",
      "id": "test_id",
      "image": "test_image_url",
      "title": "test_title",
    },
    "body": "## Some_title_1

### Some_title_2

Some_simple_text
",
    "bodyBegin": 9,
    "frontmatter": "id: test_id
title: test_title
date: '2020-04-05T00:00:00.000Z'
image: test_image_url
description: test_description",
    "html": "<h2 id=\\"some_title_1\\">Some<em>title</em>1</h2>
<h3 id=\\"some_title_2\\">Some<em>title</em>2</h3>
<p>Some<em>simple</em>text</p>",
  },
]
`;
